solidity/contracts/bridge/WalletCoordinator.sol

```diff
@@ -487,8 +494,36 @@ contract WalletCoordinator is OwnableUpgradeable, Reimbursable {
         }
     }
 
+    /// @notice Validates the sweep tx fee by checking if the part of the fee
+    ///         incurred by each deposit does not exceed the maximum value
+    ///         allowed by the Bridge. This function is heavily based on
+    ///         `DepositSweep.depositSweepTxFeeDistribution` function.
+    /// @param sweepTxFee The sweep transaction fee.
+    /// @param depositsCount Count of the deposits swept by the sweep transaction.
+    /// @dev Requirements:
+    ///      - The sweep tx fee must be grater than zero,
+    ///      - The maximum per-deposit sweep tx fee must be lesser than or equal
+    ///        the maximum fee allowed by the Bridge (`Bridge.depositTxMaxFee`).
+    function validateSweepTxFee(uint256 sweepTxFee, uint256 depositsCount)
+        internal
+        view
+    {
+        require(sweepTxFee > 0, "Proposed transaction fee cannot be zero");
+
+        uint256 depositTxFeeRemainder = sweepTxFee % depositsCount;
+        uint256 depositTxFee = (sweepTxFee - depositTxFeeRemainder) /
+            depositsCount;
+
+        (, , uint64 depositTxMaxFee, ) = bridge.depositParameters();
+
+        require(
+            depositTxFee + depositTxFeeRemainder <= depositTxMaxFee,
+            "Proposed transaction fee is too high"
+        );
```

Comment:
I am confused about why do we add `depositTxFeeRemainder`. 

`depositTxMaxFee` is a per-deposit fee:

https://github.com/keep-network/tbtc-v2/blob/9429ad99a49ac652ead710bdc2654cecd3e97cb1/solidity/contracts/bridge/Bridge.sol#L1659-L1662

Say the proposed `sweepTxFee` is 11 and there are 5 deposits.

`depositTxFeeRemainder = 1`
`depositTxFe = 2`

Then we check whether `2 + 1 <= depositTxMaxFee` but we do not have that `1` remainder left for all deposits in the sweep batch.

Shouldn't we just take the floor (`sweepTxFee / depositsCount`) and check if it's `<= depositTxMaxFee` ?

Reply:
The remainder is incurred by the last deposit so `depositTxFee + depositTxFeeRemainder` is the actual maximum per-deposit fee used within the given sweep. We also do the same within `DepositSweep.submitDepositSweepProof`: https://github.com/keep-network/tbtc-v2/blob/9429ad99a49ac652ead710bdc2654cecd3e97cb1/solidity/contracts/bridge/DepositSweep.sol#L203

Reply:
That makes sense. Can we copy the code comments from `depositSweepTxFeeDistribution` and just before we execute the `require` explain that:

```
// The transaction fee is incurred by each deposit evenly except for the last
// deposit that has the indivisible remainder additionally incurred. 
// See `DepositSweep.submitDepositSweepProof`.
// We must make sure the highest value of the deposit transaction fee does
// not exceed the maximum value limited by the governable parameter.
```

Reply:
Can we also add comments to the specific lines? Those in `depositSweepTxFeeDistribution` helps to understand the logic behind modulo:

https://github.com/keep-network/tbtc-v2/blob/9429ad99a49ac652ead710bdc2654cecd3e97cb1/solidity/contracts/bridge/DepositSweep.sol#L564-L572

Reply:
https://github.com/keep-network/tbtc-v2/pull/575/commits/67035d18d3e2115a916f931edb66b853ab99fa1d